### PR TITLE
Revert bundler version

### LINKED ISSUE #12818: [8.x] Bundler is trying to update /usr/share/logstash/Gemfile.lock on daemon start
Logstash: 8.x (master)
Artifact: deb
OS: Ubuntu 20.04
Problem:

After changing the ownership of /usr/share/logstash to be owned by 'root', Logstash is failing to start (in 8.x only) because of the following error. Looking at the error, it would seem as if bundler is attempting to write to /usr/share/logstash and failing.

```
Apr 14 09:55:31 primary logstash[50299]: `/usr/share/logstash` is not writable.
Apr 14 09:55:31 primary logstash[50299]: Bundler will use `/tmp/bundler20210414-50299-hfral650299' as your home directory temporarily.
Apr 14 09:55:39 primary logstash[50299]: Resolving dependencies..........................................................................................................................................................................................................................................................................................
Apr 14 09:55:40 primary logstash[50299]: [FATAL] 2021-04-14 09:55:40.808 [main] Logstash - Logstash stopped processing because of an error: (PermissionError) Bundler::PermissionError
Apr 14 09:55:40 primary logstash[50299]: org.jruby.exceptions.StandardError: (PermissionError) Bundler::PermissionError
Apr 14 09:55:40 primary logstash[50299]:         at usr.share.logstash.vendor.bundle.jruby.$2_dot_5_dot_0.gems.bundler_minus_2_dot_2_dot_16.lib.bundler.shared_helpers.filesystem_access(/usr/share/logstash/vendor/bundle/jruby/2.5.0/gems/bundler-2.2.16/lib/bundler/shared_helpers.rb:105) ~[?:?]
Apr 14 09:55:40 primary logstash[50299]:         at usr.share.logstash.vendor.bundle.jruby.$2_dot_5_dot_0.gems.bundler_minus_2_dot_2_dot_16.lib.bundler.definition.lock(/usr/share/logstash/vendor/bundle/jruby/2.5.0/gems/bundler-2.2.16/lib/bundler/definition.rb:369) ~[?:?]
Apr 14 09:55:40 primary logstash[50299]:         at usr.share.logstash.vendor.bundle.jruby.$2_dot_5_dot_0.gems.bundler_minus_2_dot_2_dot_16.lib.bundler.runtime.lock(/usr/share/logstash/vendor/bundle/jruby/2.5.0/gems/bundler-2.2.16/lib/bundler/runtime.rb:104) ~[?:?]
Apr 14 09:55:40 primary logstash[50299]:         at usr.share.logstash.vendor.bundle.jruby.$2_dot_5_dot_0.gems.bundler_minus_2_dot_2_dot_16.lib.bundler.runtime.setup(/usr/share/logstash/vendor/bundle/jruby/2.5.0/gems/bundler-2.2.16/lib/bundler/runtime.rb:41) ~[?:?]
Apr 14 09:55:40 primary logstash[50299]:         at usr.share.logstash.vendor.bundle.jruby.$2_dot_5_dot_0.gems.bundler_minus_2_dot_2_dot_16.lib.bundler.setup(/usr/share/logstash/vendor/bundle/jruby/2.5.0/gems/bundler-2.2.16/lib/bundler.rb:148) ~[?:?]
Apr 14 09:55:40 primary logstash[50299]:         at usr.share.logstash.lib.bootstrap.bundler.setup!(/usr/share/logstash/lib/bootstrap/bundler.rb:77) ~[?:?]
Apr 14 09:55:40 primary logstash[50299]:         at usr.share.logstash.lib.bootstrap.environment.<main>(/usr/share/logstash/lib/bootstrap/environment.rb:84) ~[?:?]
Apr 14 09:55:41 primary logstash[50619]: Using bundled JDK: /usr/share/logstash/jdk
Apr 14 09:55:41 primary logstash[50619]: OpenJDK 64-Bit Server VM warning: Option UseConcMarkSweepGC was deprecated in version 9.0 and will likely be removed in a future release.
```

If, however, we make /usr/share/logstash  writable by the daemon, it would update Gemlock.file to add a line as per the following:

```
diff --git a/Gemfile.lock b/Gemfile.lock
index a54ceff5..546b5696 100644
--- a/Gemfile.lock
+++ b/Gemfile.lock
@@ -675,6 +675,7 @@ GEM
 
 PLATFORMS
   universal-java-1.8
+  universal-java-11
 
 DEPENDENCIES
   atomic (~> 1)
```

This means that when Logstash daemon is started, it is causing bundler to run and try to update the Gemfile.lock. This should never happen because bundler shouldn't be attempting to update anything, unless instructed to do so. This issue doesn't happen in 7.x.

### PR DESCRIPTION
**Pull request by robbavey**
Recent versions of bundler (> 2.2.3) have changed the resolution to resolve
for the specific running platform, using the specific platform triple, such
as `universal-java-1.8`, rather than `java`, as was previously the case, meaning
that the 'PLATFORMS` section of the `Gemfile.lock` will change from:

```
PLATFORMS
java
```

after:

```
PLATFORMS
universal-java-1.8
```

At runtime, unless bundler is run in frozen mode, this value is resolved again at
runtime, and depending on whether the platform being run in matches that being built
on, `Gemfile.lock` is resolved to match the running platform - which now causes a
runtime error when running Logstash as a service, as the running application no longer
has write permission on `Gemfile.lock` after #12782.

While this should fix #12818, we should look into a permanent solution to enable us to
keep up with bundler updates.

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files (and/or docker env variables)
- [ ] I have added tests that prove my fix is effective or that my feature works

## Author's Checklist

- [ ]

## Related issues

-

### CODE REVIEW

**Review by yaauie — APPROVED** [review]
LGTM📍(pending 💚)

File: build.gradle [hunk 1]
```diff
@@ -204,14 +204,14 @@ tasks.register("installBundler") {
...
-      gem(projectDir, buildDir, "bundler", "~> 2", "${projectDir}/vendor/bundle/jruby/2.5.0")
+      gem(projectDir, buildDir, "bundler", "~> 2.1.4", "${projectDir}/vendor/bundle/jruby/2.5.0")
   }
 }
 
 tasks.register("bootstrap"){
     dependsOn installBundler
     doLast {
-      setupJruby(projectDir, buildDir)
+      setupJruby(projectDir, buildDir.1.4)
```
Comment (yaauie) [inline build.gradle:214]:
```suggestion
      setupJruby(projectDir, buildDir)
```

**Review by yaauie — CHANGES_REQUESTED** [review]

**Comment by robbavey** [comment]
@yauuie - ready for re-review

**Comment by robbavey** [comment]
https://github.com/elastic/logstash/pull/13015 removes the need for this PR

**Pull request closed without merging**